### PR TITLE
perf(core/mdn-annotation): avoid request to specmap.json

### PR DESCRIPTION
**Pull request by sidvishnoi**
... and added typescript annotations

There is a scope for a large number of refactor/cleanups in this module. Should I do them here or separate PRs?

### CODE REVIEW

File: src/core/mdn-annotation.js [hunk 1]
```diff
@@ -189,3 +171,33 @@ function getMdnKey(conf) {
...
+/**
+ * @param {string} key MDN key
+ * @param {object} mdnConf
+ * @param {string} [mdnConf.specMapUrl]
+ * @param {string} [mdnConf.baseJsonPath]
+ * @param {number} [mdnConf.maxAge]
+ *
+ * @typedef {{ [browser in keyof MDN_BROWSERS]: { version_added: string } }} MdnSupportEntry
+ * @typedef {{ name: string, title: string, summary: string, support: MdnSupportEntry }} MdnEntry
+ * @typedef {{ [id: string]: MdnEntry[] }} MdnData
```
Comment (saschanaz) [inline src/core/mdn-annotation.js:184]:
```suggestion
 * @typedef {Record<string, MdnEntry[]>} MdnData
```

File: src/core/mdn-annotation.js [hunk 2]
```diff
@@ -189,3 +171,33 @@ function getMdnKey(conf) {
...
 }
+
+/**
+ * @param {string} key MDN key
+ * @param {object} mdnConf
+ * @param {string} [mdnConf.specMapUrl]
+ * @param {string} [mdnConf.baseJsonPath]
+ * @param {number} [mdnConf.maxAge]
+ *
+ * @typedef {{ [browser in keyof MDN_BROWSERS]: { version_added: string } }} MdnSupportEntry
```
Comment (saschanaz) [inline src/core/mdn-annotation.js:182]:
```suggestion
 * @typedef {Record<keyof MDN_BROWSERS, { version_added: string }>} MdnSupportEntry
```

Comment (sidvishnoi) [reply src/core/mdn-annotation.js:182]: I kinda prefer this to `Record<>`, as we can specify what the key is (`browser` as a string here).

Comment (saschanaz) [reply src/core/mdn-annotation.js:182]: Well, I think `keyof MDN_BROWSERS` already is a useful hint in this case 😉

Comment (sidvishnoi) [reply src/core/mdn-annotation.js:181]: Good point!

File: src/core/mdn-annotation.js [hunk 3]
```diff
@@ -189,3 +170,31 @@ function getMdnKey(conf) {
...
+/**
+ * @param {string} key MDN key
+ * @param {object} mdnConf
+ * @param {string} [mdnConf.specMapUrl]
+ * @param {string} [mdnConf.baseJsonPath]
+ * @param {number} [mdnConf.maxAge]
+ *
+ * @typedef {Record<keyof MDN_BROWSERS, { version_added: string }>} MdnSupportEntry
+ * @typedef {{ name: string, title: string, summary: string, support: MdnSupportEntry }} MdnEntry
+ * @typedef { Record<string, MdnEntry[]>} MdnData
```
Comment (saschanaz) [inline src/core/mdn-annotation.js:183]:
```suggestion
 * @typedef {Record<string, MdnEntry[]>} MdnData
```

**Pull request merged**